File: auth/src/main/java/com/diagorn/sparkathon/auth/config/properties/JwtProperties.java
package com.diagorn.sparkathon.auth.config.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

/**
 * JWT properties
 *
 * @author diagorn
 */
@Component
@ConfigurationProperties(prefix = "application.jwt")
@Data
public class JwtProperties {
    /**
     * Access token expire time, seconds
     */
    private Integer accessExpireTimeSec;
    /**
     * Refresh token expire time, seconds
     */
    private Integer refreshExpireTimeSec;

    private String secret;
}

File: auth/src/main/java/com/diagorn/sparkathon/auth/config/properties/JwtProperties.kt
package com.diagorn.sparkathon.auth.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * JWT properties
 *
 * @author diagorn
 */
@ConfigurationProperties(prefix = "application.jwt")
class JwtProperties(
    /**
     * Access token expire time, seconds
     */
    val accessExpireTimeSec: Int,

    /**
     * Refresh token expire time, seconds
     */
    val refreshExpireTimeSec: Int,

    val secret: String
)